Client/Ecosia/Analytics/Analytics.swift

```diff
@@ -147,6 +147,12 @@ final class Analytics {
             .label("push_notification_consent")
             .property(Property.home.rawValue)
         
+        // When the user sees the APNConsent
+        // we add the number of search counts as value of the event
+        if action == .view {
+            _ = event.value(NSNumber(integerLiteral: User.shared.searchCount))
```

Comment:
Isn't it necessary to reassign `event` because the `value` function returns the updated self? Or does it already update the original copy? Might be safer and more readable to just update the value directly since its public.

```suggestion
            event.value = NSNumber(integerLiteral: User.shared.searchCount)
```

Reply:
Thanks for the suggestion @lucaschifino!
Given the `Structured` event being a class, we get all the pros (in this scenario) of the constant `let event` as reference type (so no copy).
The `func value(_ value: NSNumber?) -> Self` I'm calling at line 153 sets the `value` property and doesn't return a new copy of it but the same instance.
Hence I the `_ =` to purely silence the warning.
However, I pushed another version that just sets the `event.value` which you might like more 👌 
Let me know 💪

Reply:
Awesome, you're totally right it would still update the original reference, but I do agree setting the value is more readable, thanks!